p4-16/psa/psa.p4

```diff
@@ -83,15 +135,64 @@ const   CloneSessionId_t PSA_CLONE_SESSION_TO_CPU = unspecified;
  * fields may be used in a single P4 header definition, even on P4
  * implementations that restrict headers to contain fields with a
  * total length that is a multiple of 8 bits. */
-typedef bit<32> PortIdInHeader_t;
-typedef bit<32> MulticastGroupInHeader_t;
-typedef bit<16> CloneSessionIdInHeader_t;
-typedef bit<8>  ClassOfServiceInHeader_t;
-typedef bit<16> PacketLengthInHeader_t;
-typedef bit<16> EgressInstanceInHeader_t;
-typedef bit<64> TimestampInHeader_t;
+@p4runtime_translation("p4.org/psa/v1/PortIdInHeader_t", 32)
+type bit<32> PortIdInHeader_t;
+type bit<32> MulticastGroupInHeader_t;
+type bit<16> CloneSessionIdInHeader_t;
+@p4runtime_translation("p4.org/psa/v1/ClassOfServiceInHeader_t", 8)
+type bit<8>  ClassOfServiceInHeader_t;
+type bit<16> PacketLengthInHeader_t;
+type bit<16> EgressInstanceInHeader_t;
+type bit<64> TimestampInHeader_t;
+
+/* See the comments near the definition of PortIdUint_t for why these
+ * typedef definitions exist. */
+typedef bit<32> PortIdInHeaderUint_t;
```

Comment:
Maybe you want to move these before the `type`s and use them there.
`type PortIdInHeaqderUint_t PortIdInHeader_t;`
One less chance to get it wrong.

Reply:
Nice idea.  I tried it, but latest p4test/p4c give error when trying to use a value of such a `type` in a struct.  See https://github.com/p4lang/p4c/issues/1535

Reply:
Leaving this unchanged for now, but will consider using the idea when the bug fix is merged in, probably soon-ish.

Reply:
I have made a commit after you approved, Calin, but it was only to follow this suggestion of Mihai's to define the typedef types first, then after that define the `type` types based on the `typedef` types, so they automatically have the same width as each other without having the number in the file twice.

I also updated a few comments in the psa.p4 file, I believe accurately, but you may want to take a quick look at just the changes I made in the most recent commit on 2018-Oct-05 to double check.